### PR TITLE
Regenerate

### PR DESCRIPTION
**Pull request by alatiera**

### CODE REVIEW

File: gdk-pixbuf/src/auto/versions.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 Generated by gir (https://github.com/gtk-rs/gir @ bad7311422ee)
-from gir-files (https://github.com/gtk-rs/gir-files @ 7ebd4478b4a5)
+from gir-files (https://github.com/gtk-rs/gir-files @ 38e847e61935+)
```
Comment (alatiera) [inline gdk-pixbuf/src/auto/versions.txt:2]: Depends on https://github.com/gtk-rs/gir-files/pull/116 before we fix the dirty submodule

File: pango/src/auto/font_family.rs [hunk 2]
```diff
@@ -43,6 +61,26 @@ pub trait FontFamilyExt: 'static {
 
     #[doc(alias = "pango_font_family_list_faces")]
     fn list_faces(&self) -> Vec<FontFace>;
+
+    #[cfg(any(feature = "v1_52", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v1_52")))]
+    #[doc(alias = "item-type")]
+    fn item_type(&self) -> glib::types::Type;
```
Comment (sdroege) [inline pango/src/auto/font_family.rs:68]: See https://github.com/gtk-rs/gtk4-rs/pull/1056#discussion_r899447699

File: gdk-pixbuf/src/auto/versions.txt (hunk 1 above)
Comment (alatiera) [reply gdk-pixbuf/src/auto/versions.txt:2]: Split into https://github.com/gtk-rs/gir-files/pull/117

**Comment by sdroege** [comment]
This needs to be updated now. New gir and gir-files, and those useless new properties need to be ignored individually.

**Comment by alatiera** [comment]
Updated gir and regenerated. Ignoring the props also got rid of the pango updates which would have been blocked on the upstream MR else.

File: gio/src/auto/functions.rs [hunk 3]
```diff
@@ -467,6 +467,13 @@ pub fn io_error_from_errno(err_no: i32) -> IOErrorEnum {
     unsafe { from_glib(ffi::g_io_error_from_errno(err_no)) }
 }
 
+//#[cfg(any(feature = "v2_74", feature = "dox"))]
+//#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+//#[doc(alias = "g_io_error_from_file_error")]
```
Comment (alatiera) [inline gio/src/auto/functions.rs:472]: oh missed this one

File: gio/src/auto/flags.rs [hunk 4]
```diff
@@ -2071,6 +2071,8 @@ impl ToValue for SubprocessFlags {
 bitflags! {
     #[doc(alias = "GTlsCertificateFlags")]
     pub struct TlsCertificateFlags: u32 {
+        #[doc(alias = "G_TLS_CERTIFICATE_FLAGS_NONE")]
+        const FLAGS_NONE = ffi::G_TLS_CERTIFICATE_FLAGS_NONE as u32;
```
Comment (sdroege) [inline gio/src/auto/flags.rs:2075]: Can be hidden. This is the same as `TlsCertificateFlags::empty()`. It's useless C convenience

File: glib/gobject-sys/src/lib.rs [hunk 5]
```diff
@@ -2505,6 +2515,9 @@ extern "C" {
         strict_validation: gboolean,
     ) -> gboolean;
     pub fn g_param_value_defaults(pspec: *mut GParamSpec, value: *const GValue) -> gboolean;
+    #[cfg(any(feature = "v2_74", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+    pub fn g_param_value_is_valid(pspec: *mut GParamSpec, value: *const GValue) -> gboolean;
```
Comment (sdroege) [inline glib/gobject-sys/src/lib.rs:2520]: Needs manual implementation on `ParamSpec`: `fn value_is_valid(&self, value: &Value) -> bool`

File: glib/src/auto/functions.rs [hunk 6]
```diff
@@ -534,6 +534,20 @@ pub fn hostname_to_unicode(hostname: &str) -> Option<crate::GString> {
     unsafe { from_glib_full(ffi::g_hostname_to_unicode(hostname.to_glib_none().0)) }
 }
 
+#[cfg(any(feature = "v2_74", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+#[doc(alias = "g_idle_add_once")]
+pub fn idle_add_once<P: Fn() + Send + Sync + 'static>(function: P) -> u32 {
```
Comment (sdroege) [inline glib/src/auto/functions.rs:540]: Needs to be implemented manually, see how things are done for `g_idle_add()` in `src/source.rs`

File: glib/src/auto/functions.rs [hunk 7]
```diff
@@ -823,6 +837,26 @@ pub fn spawn_command_line_async(
 //    unsafe { TODO: call ffi:g_spawn_sync() }
 //}
 
+#[cfg(any(feature = "v2_74", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+#[doc(alias = "g_timeout_add_once")]
+pub fn timeout_add_once<P: Fn() + Send + Sync + 'static>(interval: u32, function: P) -> u32 {
```
Comment (sdroege) [inline glib/src/auto/functions.rs:843]: Needs to be implemented manually, see how things are done for `g_idle_add()` in `src/source.rs`

**Review by sdroege — COMMENTED** [review]
Seems good otherwise, thanks!

File: gio/src/auto/functions.rs [hunk 8]
```diff
@@ -467,6 +467,13 @@ pub fn io_error_from_errno(err_no: i32) -> IOErrorEnum {
     unsafe { from_glib(ffi::g_io_error_from_errno(err_no)) }
 }
 
+#[cfg(any(feature = "v2_74", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+#[doc(alias = "g_io_error_from_file_error")]
+pub fn io_error_from_file_error(file_error: glib::FileError) -> IOErrorEnum {
```
[the remark this answers is not in the document]
Comment (bilelmoussaoui) [reply gio/src/auto/functions.rs:473]: > Also maybe we should rename this to IOError instead of IOErrorEnum... @bilelmoussaoui ?

Makes sense to me

[the remark this answers is not in the document]
Comment (sdroege) [reply gio/src/auto/functions.rs:473]: Let's do that in a separate PR though (@alatiera please create an issue). The `From` impl should be done here

File: gio/src/auto/file.rs [hunk 9]
```diff
@@ -103,6 +103,158 @@ impl File {
...
+
+    #[cfg(any(feature = "v2_74", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_74")))]
+    pub fn new_tmp_future(
+        tmpl: Option<impl AsRef<std::path::Path>>,
+        io_priority: glib::Priority,
+    ) -> Pin<
+        Box_<dyn std::future::Future<Output = Result<(File, FileIOStream), glib::Error>> + 'static>,
+    > {
+        let tmpl = tmpl.map(|tmpl| tmpl.as_ref().to_owned());
```
Comment (sdroege) [inline gio/src/auto/file.rs:167]: Don't edit autogenerated files. You'll need to make this one manual, or fix gir.

Comment (alatiera) [reply gio/src/auto/file.rs:167]: Yea I know, It's easier to get the diff that way to file the issue. I will make it manual in a bit.

File: gio/src/error.rs [hunk 10]
```diff
@@ -1,5 +1,7 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
+use crate::glib::translate::from_glib;
```
Comment (bilelmoussaoui) [inline gio/src/error.rs:3]:
```suggestion
use crate::glib::translate::*;
```
you need into_glib as well, also those two new imports needs a version guard

File: glib/src/auto/functions.rs (hunk 6 above)
Comment (alatiera) [reply glib/src/auto/functions.rs:540]: slight issue, we already have a manual `idle_add_once` with which is aliasing `#[doc(alias = "g_timeout_add_full")]`

Comment (sdroege) [reply glib/src/auto/functions.rs:540]: You can just mark it `manual`, add the alias and leave it at that

File: pango/Gir.toml [hunk 11]
```diff
@@ -272,9 +271,25 @@ concurrency = "send+sync"
...
 
+[[object]]
+name = "Pango.FontFamily"
+status = "generate"
+    [[object.property]]
+    name = "item-type"
+    generate = [] # disable generated getter/notify, use the ones from GListModel iface
+    [[object.property]]
+    name = "n-items"
+    generate = [] # disable generated getter/notify, use the ones from GListModel iface
```
Comment (bilelmoussaoui) [inline pango/Gir.toml:282]: we should keep the notify signal here & for other listmodel types

File: glib/gobject-sys/src/lib.rs (hunk 5 above)
Comment (alatiera) [reply glib/gobject-sys/src/lib.rs:2520]: Done, though not sure how correct it is

**Comment by alatiera** [comment]
There are a handful warnings, do I need to do anything about them?

```
   Compiling glib v0.16.0 (/home/alatiera/Projects/gtk-rs-core/glib)
warning: associated function is never used: `value_is_valid`
   --> glib/src/param_spec.rs:150:8
    |
150 |     fn value_is_valid(&self, value: &Value) -> bool {
    |        ^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

   Compiling cairo-rs v0.16.0 (/home/alatiera/Projects/gtk-rs-core/cairo)
   Compiling pango v0.16.0 (/home/alatiera/Projects/gtk-rs-core/pango)
   Compiling graphene-rs v0.16.0 (/home/alatiera/Projects/gtk-rs-core/graphene)
warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
   --> gio/src/auto/mod.rs:505:22
    |
505 | pub use self::enums::TlsRehandshakeMode;
    |                      ^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated static `auto::constants::VOLUME_IDENTIFIER_KIND_HAL_UDI`: Since 2.58
   --> gio/src/auto/mod.rs:679:26
    |
679 | pub use self::constants::VOLUME_IDENTIFIER_KIND_HAL_UDI;
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `glib` (lib) generated 1 warning
warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
  --> gio/src/auto/tls_connection.rs:15:12
   |
15 | use crate::TlsRehandshakeMode;
   |            ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
  --> gio/src/auto/tls_connection.rs:90:35
   |
90 |     fn rehandshake_mode(&self) -> TlsRehandshakeMode;
   |                                   ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
   --> gio/src/auto/tls_connection.rs:128:42
    |
128 |     fn set_rehandshake_mode(&self, mode: TlsRehandshakeMode);
    |                                          ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
   --> gio/src/auto/tls_connection.rs:280:35
    |
280 |     fn rehandshake_mode(&self) -> TlsRehandshakeMode {
    |                                   ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
   --> gio/src/auto/tls_connection.rs:415:42
    |
415 |     fn set_rehandshake_mode(&self, mode: TlsRehandshakeMode) {
    |                                          ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3785:23
     |
3785 | impl fmt::Display for TlsRehandshakeMode {
     |                       ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3801:19
     |
3801 | impl IntoGlib for TlsRehandshakeMode {
     |                   ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3815:45
     |
3815 | impl FromGlib<ffi::GTlsRehandshakeMode> for TlsRehandshakeMode {
     |                                             ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3826:21
     |
3826 | impl StaticType for TlsRehandshakeMode {
     |                     ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3832:33
     |
3832 | impl glib::value::ValueType for TlsRehandshakeMode {
     |                                 ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3836:35
     |
3836 | unsafe impl<'a> FromValue<'a> for TlsRehandshakeMode {
     |                                   ^^^^^^^^^^^^^^^^^^

warning: use of deprecated enum `auto::enums::TlsRehandshakeMode`: Since 2.60
    --> gio/src/auto/enums.rs:3844:18
     |
3844 | impl ToValue for TlsRehandshakeMode {
     |                  ^^^^^^^^^^^^^^^^^^

   Compiling pangocairo v0.16.0 (/home/alatiera/Projects/gtk-rs-core/pangocairo)
warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Never`: Since 2.60
    --> gio/src/auto/enums.rs:3791:23
     |
3791 |                 Self::Never => "Never",
     |                       ^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Safely`: Since 2.60
    --> gio/src/auto/enums.rs:3792:23
     |
3792 |                 Self::Safely => "Safely",
     |                       ^^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Unsafely`: Since 2.60
    --> gio/src/auto/enums.rs:3793:23
     |
3793 |                 Self::Unsafely => "Unsafely",
     |                       ^^^^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Never`: Since 2.60
    --> gio/src/auto/enums.rs:3806:19
     |
3806 |             Self::Never => ffi::G_TLS_REHANDSHAKE_NEVER,
     |                   ^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Safely`: Since 2.60
    --> gio/src/auto/enums.rs:3807:19
     |
3807 |             Self::Safely => ffi::G_TLS_REHANDSHAKE_SAFELY,
     |                   ^^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Unsafely`: Since 2.60
    --> gio/src/auto/enums.rs:3808:19
     |
3808 |             Self::Unsafely => ffi::G_TLS_REHANDSHAKE_UNSAFELY,
     |                   ^^^^^^^^

warning: use of deprecated tuple variant `auto::enums::TlsRehandshakeMode::__Unknown`: Since 2.60
    --> gio/src/auto/enums.rs:3809:19
     |
3809 |             Self::__Unknown(value) => value,
     |                   ^^^^^^^^^

warning: use of deprecated field `auto::enums::TlsRehandshakeMode::__Unknown::0`: Since 2.60
    --> gio/src/auto/enums.rs:3809:29
     |
3809 |             Self::__Unknown(value) => value,
     |                             ^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Never`: Since 2.60
    --> gio/src/auto/enums.rs:3818:51
     |
3818 |             ffi::G_TLS_REHANDSHAKE_NEVER => Self::Never,
     |                                                   ^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Safely`: Since 2.60
    --> gio/src/auto/enums.rs:3819:52
     |
3819 |             ffi::G_TLS_REHANDSHAKE_SAFELY => Self::Safely,
     |                                                    ^^^^^^

warning: use of deprecated unit variant `auto::enums::TlsRehandshakeMode::Unsafely`: Since 2.60
    --> gio/src/auto/enums.rs:3820:54
     |
3820 |             ffi::G_TLS_REHANDSHAKE_UNSAFELY => Self::Unsafely,
     |                                                      ^^^^^^^^

warning: use of deprecated tuple variant `auto::enums::TlsRehandshakeMode::__Unknown`: Since 2.60
    --> gio/src/auto/enums.rs:3821:28
     |
3821 |             value => Self::__Unknown(value),
     |                            ^^^^^^^^^

   Compiling gdk-pixbuf v0.16.0 (/home/alatiera/Projects/gtk-rs-core/gdk-pixbuf)
warning: `gio` (lib) generated 26 warning
```

**Comment by sdroege** [comment]
> warning: associated function is never used: `value_is_valid`

Add a `pub` in front of the function.

The others are all expected.

**Review by sdroege — DISMISSED** [review]
Good for you @bilelmoussaoui ?

File: glib/Gir.toml [hunk 12]
```diff
@@ -379,12 +380,18 @@ status = "generate"
     name = "idle_add_full"
     ignore = true
     [[object.function]]
+    name = "idle_add_once"
+    ignore = true
+    [[object.function]]
```
Comment (bilelmoussaoui) [inline glib/Gir.toml:385]: could use a small comment saying we have that manually implemented? same for timeout_add_once. Maybe even mark them as manual so we can profit from the upstream docs?

File: glib/Gir.toml [hunk 13]
```diff
@@ -379,12 +380,18 @@ status = "generate"
...
     name = "timeout_add"
     ignore = true
     [[object.function]]
     name = "timeout_add_full"
     ignore = true
     [[object.function]]
+    name = "timeout_add_once"
```
Comment (bilelmoussaoui) [inline glib/Gir.toml:392]: same for this

File: glib/src/functions.rs [hunk 14]
```diff
@@ -231,7 +231,7 @@ pub fn unix_open_pipe(flags: i32) -> Result<(RawFd, RawFd), Error> {
     unsafe {
         let mut fds = [0, 2];
         let mut error = ptr::null_mut();
-        let _ = ffi::g_unix_open_pipe(fds.as_mut_ptr(), flags, &mut error);
+        let _ = ffi::g_unix_open_pipe(&mut fds, flags, &mut error);
```
Comment (bilelmoussaoui) [inline glib/src/functions.rs:234]: what was the reason of this change? could be split in a separate commit

File: pango/Gir.toml [hunk 15]
```diff
@@ -272,9 +271,25 @@ concurrency = "send+sync"
     name = "merge_static"
     ignore = true
 
+[[object]]
+name = "Pango.FontFamily"
+status = "generate"
+    [[object.property]]
```
Comment (bilelmoussaoui) [inline pango/Gir.toml:277]: I find it very weird how a ListModel like properties were added but that gir didn't add a `implements GListModel` to glib-wrapper call

File: glib/src/functions.rs (hunk 14 above)
Comment (sdroege) [reply glib/src/functions.rs:234]: The gir file changed so the parameter of the -sys function changed from `*mut i32` to `[i32; 2]`

File: pango/Gir.toml (hunk 15 above)
Comment (sdroege) [reply pango/Gir.toml:277]: You need to add `Gio.ListModel` to the `manual` list for that to work

File: glib/src/functions.rs (hunk 14 above)
Comment (bilelmoussaoui) [reply glib/src/functions.rs:234]: ah right

File: glib/Gir.toml (hunk 12 above)
Comment (alatiera) [reply glib/Gir.toml:385]: They have handwritten doc comments, also deleting the comment and switching it to manual doesn't seem to produce any doc either for some reason.

File: pango/Gir.toml [hunk 16]
```diff
@@ -55,6 +54,7 @@ manual = [
     "GLib.Bytes",
     "GLib.Error",
     "GLib.Quark",
+    "Gio.ListModel",
```
Comment (sdroege) [inline pango/Gir.toml:57]: `Gio` < `GLib` :P

**Review by bilelmoussaoui — APPROVED** [review]
